tests/unit/test_model.py

```diff
@@ -251,27 +251,28 @@ async def test_no_args(self):
             # no apps so should return right away
             await m.wait_for_idle(wait_for_active=True)
 
-    async def test_apps_no_lst(self):
+    async def test_apps_type(self):
         m = Model()
-        with self.assertRaises(JujuError):
+        with self.assertRaises(TypeError):
             # apps arg has to be a List[str]
             await m.wait_for_idle(apps="should-be-list")
 
-        with self.assertRaises(JujuError):
+        with self.assertRaises(TypeError):
             # apps arg has to be a List[str]
-            await m.wait_for_idle(apps=3)
+            await m.wait_for_idle(apps=3)  # type: ignore
 
-        with self.assertRaises(JujuError):
+        with self.assertRaises(TypeError):
             # apps arg has to be a List[str]
```

Comment:
I remember noticing this change before, though I'm not sure we talked about it -- is there any benefit to changing the error type here?

Reply:
Only that it's clearer where the error originates: client-side or Juju.

Reply:
That's a good enough reason for me =)